### DISCUSSION TITLE
Cant build my app anymore after PR 1495

### DISCUSSION THREAD

**Discussion by Skagevang** [TanStack/router#1521] [Q&A] [answered]
Why implement the "feature" of only allow leaf nodes? From [this pr](https://github.com/TanStack/router/pull/1495). Now my app wont build anymore because of typescript errors. Should we only be allowed to route to leaf nodes? 

![image](https://github.com/TanStack/router/assets/11090538/affa97f9-14d8-43d0-9ab6-d3f2d3063ac7)
The image above shows one of my folders. According to typescript I cant navigate to /ilp/$studentId because there also is a nested route   of $studentId. 

Btw this also broke [your own example](https://tanstack.com/router/latest/docs/framework/react/examples/location-masking) in the docs...

**Comment by schiller-manuel** [comment, marked as answer]
this has been loosened up in https://github.com/TanStack/router/releases/tag/v1.31.0 again, so please upgrade
just FYI, you might still need `$studentId.index.tsx` if you want to render something on `/ilp/123` that is NOT rendered on `/ilp/123/456`

**Reply by Skagevang** [reply]
Thank you! This worked and I have upgraded 😄
